opentelemetry-exporter-geneva/geneva-uploader/src/client.rs

```diff
@@ -0,0 +1,107 @@
+pub struct GenevaClient {
+    uploader: Arc<GenevaUploader>,
+    encoder: Arc<OtlpEncoder>,
+    metadata: String,
+}
+
+impl GenevaClient {
+    /// Construct a new client with minimal configuration. Fetches and caches ingestion info as needed.
+    pub async fn new(cfg: GenevaClientConfig) -> Result<Self, String> {
+        // Build config client config
+        let config_client_config = GenevaConfigClientConfig {
+            endpoint: cfg.endpoint,
+            environment: cfg.environment.clone(),
+            account: cfg.account,
+            namespace: cfg.namespace.clone(),
+            region: cfg.region,
+            config_major_version: cfg.config_major_version,
+            auth_method: cfg.auth_method,
+        };
+        let config_client = Arc::new(
+            GenevaConfigClient::new(config_client_config)
+                .map_err(|e| format!("GenevaConfigClient init failed: {e}"))?,
+        );
+
+        let source_identity = format!(
+            "Tenant={}/Role={}/RoleInstance={}",
+            cfg.tenant, cfg.role_name, cfg.role_instance
+        );
+
+        let schema_ids =
+            "c1ce0ecea020359624c493bbe97f9e80;0da22cabbee419e000541a5eda732eb3".to_string(); // TODO - find the actual value to be populated
+
+        // Uploader config
+        let uploader_config = GenevaUploaderConfig {
+            namespace: cfg.namespace.clone(),
+            source_identity,
+            environment: cfg.environment,
+            schema_ids,
+        };
+
+        let uploader = GenevaUploader::from_config_client(config_client, uploader_config)
+            .await
+            .map_err(|e| format!("GenevaUploader init failed: {e}"))?;
+        let metadata = format!(
+            "namespace={}/eventVersion={}/tenant={}/role={}/roleinstance={}",
+            cfg.namespace,
+            "Ver1v0", // You can replace this with a cfg field if version should be dynamic
+            cfg.tenant,
+            cfg.role_name,
+            cfg.role_instance,
+        );
+        Ok(Self {
+            uploader: Arc::new(uploader),
+            encoder: Arc::new(OtlpEncoder::new()),
+            metadata,
+        })
+    }
+
+    /// Upload OTLP logs (as ResourceLogs).
+    pub async fn upload_logs(&self, logs: Vec<ResourceLogs>) -> Result<(), String> {
+        let log_iter = logs
+            .iter()
+            .flat_map(|resource_log| resource_log.scope_logs.iter())
+            .flat_map(|scope_log| scope_log.log_records.iter());
+        let blobs = self.encoder.encode_log_batch(log_iter, &self.metadata);
+        for (_schema_id, event_name, encoded_blob, _row_count) in blobs {
+            // TODO - log encoded_blob for debugging
+            let compressed_blob = lz4_chunked_compression(&encoded_blob)
+                .map_err(|e| format!("LZ4 compression failed: {e}"))?;
+            // TODO - log compressed_blob for debugging
+            let event_version = "Ver2v0"; // TODO - find the actual value to be populated
```

Comment:
I see "Ver1v0" being used in `GenevaClient::new()` method. Should these two have been the same string? If yes, maybe use a const variable at both places for the time being.